pkg/addonmanager/manager.go

```diff
@@ -131,6 +131,14 @@ func (a *addonManager) Start(ctx context.Context) error {
 		workInformers.Work().V1().ManifestWorks(),
 		a.addonAgents,
 	)
+	hostingDeployController := agentdeploy.NewAddonHostingDeployController(
```

Comment:
this should not be started by default.

Reply:
Should we add a feature gate(ex: AddonAgentHostedMode)?

Reply:
not a feature gate, but a agentOption field

Reply:
fixed